subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1187,12 +1217,17 @@ static int open_storage(uint32_t offset, bool preserve)
 		LOG_ERR("Failed to init flash stream for offset %u: %d",
 			offset, err);
 		state = PGPS_NONE;
-		return err;
+		goto fail;
 	}
 
 	if (preserve && (block_offset != 0) && (block_offset < flash_page_size)) {
 		int slot = offset / PGPS_PREDICTION_STORAGE_SIZE;
 		uint8_t *buf = (uint8_t *)get_prediction_slot(slot, NULL);
```

Comment:
I think this will double-take the semaphore if CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL is enabled, right? Should it be a mutex, instead?

Reply:
I'm not sure what line your comment is about, nor where the two takes would occur.  Please elaborate.

Reply:
Never mind! I misread. I thought `get_prediction_slot` was one of the functions you added the semaphore to, but that isn't the case.